Contributing.md

```diff
@@ -0,0 +1,137 @@
+ - feat: adding a new feature
+ - fix: fixing something
+ - improvement: improve something
+ - perf: improving performance
+ - refactor: refactor code
+ - revert: undo a previous change
+ - style: working only on code or documentation style
+ - test: commits for testing code
+
+Valid scopes: the scope is optional and usually refers to which module is being changed.
+ - TBD - for now not validated, should be just one word ideally
+
+Valid subject: short, free form, what the commit is about in less than 50 or 60 characters
+(not strictly enforced, but it's best to keep it short)
+
+Optional body: this is where you put all the verbose details you want about the commit, or
+nothing at all if the subject already says it all. Must be separated by a blank line from
+the subject. Explain what the changes are, why you're doing them, etc, as necessary.
+
+Optional footer: separated from the body (or subject if body is empty) by a blank line,
+lists reference (e.g.: "Closes #12" "Ref #24") or warns of breaking changes (e.g.,
+"BREAKING CHANGE: explanation").
+
+These rules are inspired by these commit formatting guides:
+ - [Conventional Commits](https://www.conventionalcommits.org/)
+ - [Bluejava commit guide](https://github.com/bluejava/git-commit-guide)
+ - [develar's commit message format](https://gist.github.com/develar/273e2eb938792cf5f86451fbac2bcd51)
+ - [AngularJS Git Commit Message Conventions](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y).
+
+### Enabling commitlint
+
+We run commitlint on each commit message that you write by enabling the commit-msg hook in
+Git. It is run via [husky](https://www.npmjs.com/package/husky), which is a JS Git hook
+manager, and you need Node to run it.
+
+If you don't already use Node, this is a bit more work to install that the pre-commit
+hooks above, but please take a moment to do this:
+
+- If you don't already use Node or nvm, install nvm in your ~/.nvm folder:
+```sh
+wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash`
+```
+This will add a few lines to your `.bashrc` file, which you'll need to execute now,
+possibly by starting a new shell.
+
+- Install Node, preferably version 12.18.3
+```sh
+nvm install node && nvm install 12.18.3
+```
+
+- Use Node 12.18.3: you need to do this each time, so adding it to your .bashrc is a good
+  idea unless you already use other versions of Node for other projects
+```sh
+nvm use 12.18.3
+```
+
+- In your ReadAlong/Studio sandbox, install the husky commit-msg hook using npm, the node
+  package manager you just installed using nvm. The file `package.json` in Studio is what
+  tells npm to install husky as a pre-commit hook, and also what tells husky to invoke
+  commitlint on your commit messages.
+```sh
+npm install
+```
+
+- Now, next time you make a change and commit it, your commit log will be checked:
+  - `git commit -m'non-compliant commit log text'` outputs an error
```

Comment:
hehe, I would write this `git commit -m 'non-compliant commit log text'` but I noticed you doing it without the space on the command line so I think you meant to write it this way. I don't care, but maybe this is another tabs vs spaces type argument.

Reply:
i.e. we should add a pre-commit hook that fails if you write it like this `git commit -m'non-compliant commit log text'` but also fails if you write it like this `git commit -m 'non-compliant commit log text'` 😛

Reply:
Indeed, I never put a space there! I don't care either way how we show it here, although the second example, which I say works, also doesn't have that space.
Do you expect it might confuse some readers?